internal/app/init/pkg/system/services/containerd.go

```diff
@@ -52,11 +52,11 @@ func (c *Containerd) Start(data *userdata.UserData) error {
 		env = append(env, fmt.Sprintf("%s=%s", key, val))
 	}
 
-	r := process.Process{}
-
-	return r.Run(
+	r := process.NewRunner(
 		data,
 		args,
 		runner.WithEnv(env),
 	)
+
+	return r.Run()
```

Comment:
my idea is to return `r` eventually, but this is only possible when containerd runner is updated as well

Reply:
My plan was to make similar changes and add tests for containerd runner. I was exploring the way I can make containerd available for the tests, both in local mode and in docker build. I'm not sure if I can launch containerd while tests run as part of the build in buildkitd, but I'll try it.

Reply:
containerd does support being ran in a container. We would just need to make sure we start it with the correct storage drive because the kernel doesn't like nested overlay filesystems.